presto-hive/src/main/java/com/facebook/presto/hive/HiveConnectorThriftCodecProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.drift.codec.ThriftCodecManager;
+import com.facebook.presto.spi.ConnectorSpecificThriftCodec;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.connector.ConnectorSpecificThriftCodecProvider;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+
+import javax.inject.Inject;
+
+import static java.util.Objects.requireNonNull;
+
+public class HiveConnectorThriftCodecProvider
+        implements ConnectorSpecificThriftCodecProvider
+{
+    private final ThriftCodecManager thriftCodecManager;
+
+    @Inject
+    public HiveConnectorThriftCodecProvider(ThriftCodecManager thriftCodecManager)
+    {
+        this.thriftCodecManager = requireNonNull(thriftCodecManager, "thriftCodecManager is null");
+    }
+
+    @Override
+    public ConnectorSpecificThriftCodec<ConnectorSplit> getConnectorSplitCodec()
+    {
+        return new HiveSplitThriftCodec(thriftCodecManager);
+    }
+
+    @Override
+    public ConnectorSpecificThriftCodec<ConnectorTransactionHandle> getConnectorTransactionHandleCodec()
```

Comment:
Should this be `Optional<ConnectoThriftCodec<...>>`? In other words do we want to allow connectors provide Thrift based serialization for only a subset of entities?